packages/flutter/test/material/bottom_navigation_bar_test.dart

```diff
@@ -1976,7 +2329,63 @@ void main() {
         }
         await expectLater(
           find.byType(BottomNavigationBar),
-          matchesGoldenFile('bottom_navigation_bar.shifting_transition.${pump - 1}.png'),
+          matchesGoldenFile('m2_bottom_navigation_bar.shifting_transition.${pump - 1}.png'),
+        );
+      });
+    }
+  });
+
+  group('Material3 - BottomNavigationBar shifting backgroundColor with transition', () {
+    // Regression test for: https://github.com/flutter/flutter/issues/22226
+    Widget runTest() {
+      int currentIndex = 0;
+      const bool useInkSparkle = !kIsWeb;
+
+      return MaterialApp(
+        // Because this test relies on golden, fo a constant seed for the ink sparkle animation.
+        theme: ThemeData(splashFactory: useInkSparkle ? InkSparkle.constantTurbulenceSeedSplashFactory : null),
```

Comment:
Yes! constantTurbulenceSeedSplashFactory should eliminate the randomness we saw in the previous PR. 👍

Reply:
Just double checking - this is the only test in this PR with the sparkle?

Reply:
Yes, this is the only golden test in this PR.

For future M3 test migration PRs, this issue will surface on any golden test that simulates a tap on a widget tree that contains an `InkWell`. I don't know if there are a lot of them, the tricky part will be to be aware of this (Taha and me will take care on our migration PRs).

Reply:
Yeah, I wonder, should we create an M3 test utility for this? You could create just a simple wrapper class that creates a MaterialApp with the turbulence seed set, and then re-use it in every M3 test so we'll know it is consistently being applied.

Reply:
Oh! Actually scratch that. How about this instead - let's update InkSparkle so that in debug mode it always uses the fixed turbulence seed. That would be the best way to prevent this issue from reoccurring. Probably in the InkSparkle constructor, set the seed in an assert statement so in debug mode (including tests) it will always be fixed and consistent!

cc @TahaTesser